src/workerd/io/worker.c++

```diff
@@ -1666,7 +1667,12 @@ Worker::Worker(kj::Own<const Script> scriptParam,
                     KJ_SWITCH_ONEOF(handler.value) {
                       KJ_CASE_ONEOF(obj, api::ExportedHandler) {
                         obj.env = lock.v8Ref(bindingsScope.As<v8::Value>());
-                        obj.ctx = jsg::alloc<api::ExecutionContext>();
+                        // TODO(cleanup): Unfortunately, for non-class-based handlers, we have
+                        //   always created only a single `ctx` object and reused it for all
+                        //   requests. This is weird and obviously wrong but changing it probably
+                        //   requires a compat flag. Until then, connection properties will not be
+                        //   available for non-class handlers.
```

Comment:
For later... we *could* add a getter to `ExecutionContext` that always returns the request-specific `props` accessed through, perhaps, the `IoContext`. That's obviously a bit more work and is fairly convoluted so I'm fine with this restriction for now.

Reply:
That would actually be easy, but I think it's actually weird and broken that an `ExecutioContext` instance can be used in a different context than the one where it originated, and it'll just operate on whatever context it's used it... I mean, you would _expect_ that each context gets a unique `ExecutionContext`, which can only operate on that context. I want to fix this via a compat flag later.